src/types.ts

```diff
@@ -8,9 +8,40 @@
  */
 
 /**
- * Config accepted by the encryption
+ * The contract Encryption drivers should adhere to
  */
-export type EncryptionOptions = {
-  algorithm?: 'aes-256-cbc'
-  secret: string
+export interface EncryptionDriverContract {
+  /**
+   * Encrypt a given piece of value using the app secret. A wide range of
+   * data types are supported.
+   *
+   * - String
+   * - Arrays
+   * - Objects
+   * - Booleans
+   * - Numbers
+   * - Dates
+   *
+   * You can optionally define a purpose for which the value was encrypted and
+   * mentioning a different purpose/no purpose during decrypt will fail.
+   */
+  encrypt(payload: any, expiresIn?: string | number, purpose?: string): string
+
+  /**
+   * Decrypt value and verify it against a purpose
+   */
+  decrypt<T extends any>(value: string, purpose?: string): T | null
 }
+
+/**
+ * Factory function to return the driver implementation. The method
+ * cannot be async, because the API that calls this method is not
+ * async in first place.
+ */
+export type ManagerDriverFactory = () => EncryptionDriverContract
+
+export interface BaseConfig {
+  key: string
+}
```

Comment:
What's the purpose of `BaseConfig`?

Reply:
To have a default config that we can extends, for example, the Legacy driver is using the `BaseConfig` as-is, but other driver would be able to extend this `BaseConfig` to add their own options.